### PR TITLE
Update links; use HTTPS for links

### PR DESCRIPTION
**Pull request by Marcono1234**
Uses HTTPS for links (where supported). Additionally replaces some links which are currently redirecting with their destination.

Edit: I am not quite sure why Travis failed or why codecov thinks that the coverage has decreased.

### CODE REVIEW

File: doc/release-notes/official.md [hunk 1]
```diff
@@ -795,7 +795,7 @@
  - 2018-07-27 - [2 commits](https://github.com/mockito/mockito/compare/v2.20.1...v2.20.2) by [Rafael Winterhalter](https://github.com/raphw) (1), [Szczepan Faber](https://github.com/mockitoguy) (1) - published to [![Bintray](https://img.shields.io/badge/Bintray-2.20.2-green.svg)](https://bintray.com/mockito/maven/mockito-development/2.20.2)
  - Updates Byte Buddy to add preliminary support for Java 12. [(#1448)](https://github.com/mockito/mockito/pull/1448)
 
-# 2.20.1 (2018-07-26) published to [JCenter](https://bintray.com/mockito/maven/mockito/2.20.1)/[Maven Central](http://search.maven.org/#artifactdetails%7Corg.mockito%7Cmockito-core%7C2.20.1%7Cjar)
+# 2.20.1 (2018-07-26) published to [JCenter](https://bintray.com/mockito/maven/mockito/2.20.1)/[Maven Central](https://search.maven.org/artifact/org.mockito/mockito-core/2.20.1/jar)
```
Comment (TimvdLippe) [inline doc/release-notes/official.md:798]: Let's not update our release notes retroactively. Could you please revert the changes in this file?

**Review by TimvdLippe — CHANGES_REQUESTED** [review]

File: doc/release-notes/official.md (hunk 1 above)
Comment (Marcono1234) [reply doc/release-notes/official.md]: Reverted it in the latest commit. I have used a [fixup](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---fixupltcommitgt) commit for that, though I guess that was unnecessary since you will squash the commits anyways.

Should I also add `[skip release]` to the commit message since there are nearly no code changes?

**Review by TimvdLippe — COMMENTED** [review]
Thanks!

**Review by TimvdLippe — APPROVED** [review]
Now with approval...
